interfaces/builtin/optical_drive_test.go

```diff
@@ -77,20 +104,46 @@ func (s *OpticalDriveInterfaceSuite) TestSanitizeSlot(c *C) {
 }
 
 func (s *OpticalDriveInterfaceSuite) TestSanitizePlug(c *C) {
-	c.Assert(interfaces.BeforePreparePlug(s.iface, s.plugInfo), IsNil)
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.testPlugDefaultInfo), IsNil)
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.testPlugReadonlyInfo), IsNil)
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.testPlugWritableInfo), IsNil)
 }
 
 func (s *OpticalDriveInterfaceSuite) TestAppArmorSpec(c *C) {
-	spec := &apparmor.Specification{}
-	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slot), IsNil)
-	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
-	c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, `/dev/sr[0-9]* r,`)
+	// spec := &apparmor.Specification{}
+	// c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slot), IsNil)
+	// c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
+	// c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, `/dev/sr[0-9]* r,`)
```

Comment:
I think you can use simpler tests similar to the above with the changes I suggested.

Reply:
I've made all those changes, but locally I'm getting a single failed test - the presence of the `write` attribute is not changing the response to `autoconnect` from `true` to `false` and so the test fails:

```go
c.Assert(s.iface.AutoConnect(&interfaces.Plug{PlugInfo: s.testPlugWritableInfo}, &interfaces.Slot{SlotInfo: s.slotInfo}), Equals, false)
```